debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorTask.java

```diff
@@ -81,7 +85,7 @@ public ChangeEventSourceCoordinator<OraclePartition, OracleOffsetContext> start(
                 connectorConfig.getAdapter().getOffsetContextLoader());
 
         // Manual Bean Registration
-        beanRegistryJdbcConnection = connectionFactory.newConnection();
+        beanRegistryJdbcConnection = dualConnectionFactory.newReadonlyConnection();
```

Comment:
Why not use `newConnection()` here and consider having the dual-factory deviate whether the new connection is read-only based or using the main connection configuration?  

This way for existing users not using the new logic, there should be no underlying change in behavior.